src/test/redis_response_cache_test.cc

```diff
@@ -0,0 +1,708 @@
+    output0_size = sizeof(int) * data0.size();
+    outputs0 = std::vector<Tensor>{output_tensor0};
+    // Response of 100 ints, taking ~400 bytes at a time
+    data100 = std::vector<int>(100, 0);
+    Tensor output_tensor100 = {"output", data100};
+    outputs100 = std::vector<Tensor>{output_tensor100};
+
+    // Sample responses
+    response0 = GenerateResponse(
+        request0, dtype, memory_type, memory_type_id, outputs0);
+    ASSERT_NE(response0, nullptr);
+    response_400bytes = GenerateResponse(
+        request0, dtype, memory_type, memory_type_id, outputs100);
+    ASSERT_NE(response_400bytes, nullptr);
+  }
+
+  void TearDown() override
+  {
+    delete request0;
+    delete request1;
+    delete request2;
+    delete request3;
+    delete request4;
+    for (auto r : unique_requests) {
+      delete r;
+    }
+  }
+
+ public:
+  tc::Model* model = nullptr;
+  uint64_t model_version = 1;
+  inference::DataType dtype = inference::DataType::TYPE_INT32;
+  TRITONSERVER_MemoryType memory_type = TRITONSERVER_MEMORY_CPU;
+  int64_t memory_type_id = 0;
+  size_t thread_count = 10;
+  uint64_t output0_size;
+
+  std::vector<int> data0, data1, data100;
+  std::vector<Tensor> inputs0, inputs1, inputs2, inputs3, inputs4, inputs100;
+  std::vector<Tensor> outputs0, outputs100;
+  tc::InferenceRequest *request0, *request1, *request2, *request3, *request4;
+  std::vector<tc::InferenceRequest*> unique_requests;
+  std::unique_ptr<tc::InferenceResponse> response0, response_400bytes;
+};
+
+// Test hashing for consistency on same request
+TEST_F(RequestResponseCacheTest, TestHashing)
+{
+  // Create cache
+  std::cout << "Create cache" << std::endl;
+  uint64_t cache_size = 4 * 1024 * 1024;
+  std::unique_ptr<tc::RequestResponseCache> cache;
+  tc::RedisResponseCache::Create(
+    tc::redis_address,
+    tc::redis_username,
+    tc::redis_password, &cache);
+
+  // Compare hashes
+  std::cout << "Compare hashes" << std::endl;
+  check_status(cache->HashAndSet(request0));
+  check_status(cache->HashAndSet(request1));
+  check_status(cache->HashAndSet(request2));
+  check_status(cache->HashAndSet(request3));
+  check_status(cache->HashAndSet(request4));
+
+  std::cout << "request0->CacheKey(): " << request0->CacheKey() << std::endl;
+  std::cout << "request1->CacheKey(): " << request1->CacheKey() << std::endl;
+  std::cout << "request2->CacheKey(): " << request2->CacheKey() << std::endl;
+  std::cout << "request3->CacheKey(): " << request3->CacheKey() << std::endl;
+  std::cout << "request4->CacheKey(): " << request4->CacheKey() << std::endl;
+  // Different input data should have different hashes
+  ASSERT_NE(request0->CacheKey(), request1->CacheKey());
+  // Same input data should have same hashes
+  ASSERT_EQ(request1->CacheKey(), request2->CacheKey());
+  // Two requests with same two inputs but added in different orders
+  ASSERT_EQ(request3->CacheKey(), request4->CacheKey());
+}
+
+// TODO  ENABLE WHEN DBSIZE FIXED
```

Comment:
What will expectations and behavior be around `Eviction` ? I saw Redis advertise:
> Distributed cache of theoretically infinite size, deployed across triton instances.

Will the cache grow forever for the life of the server until OOM/storage ? Will users or Triton be expected to trigger eviction at certain points? How does Redis expect this workflow to look like?

Reply:
so traditionally in caching, you set a memory size based on the server on which the cache is placed or the ramifications of the application. Users of Triton will never have to worry about or trigger eviction strategies. the triton server itself won't even have to worry about it. All this will be handled by Redis on the server side.

Eviction strategies in redis -> https://redis.io/docs/manual/eviction/

These can all be set through API calls, cli command, admin commands, dashboards, etc